src/Ticket.php

```diff
@@ -1127,6 +1127,12 @@ public function prepareInputForUpdate($input)
             $input['itilcategories_id_code'] = ITILCategory::getById($cat_id)->fields['code'];
         }
 
+        // Set previous category code, this is needed to let the rule engine
+        // decide if the code was changed
+        if ($category = ITILCategory::getById($this->fields['itilcategories_id'] ?? 0)) {
```

Comment:
Will save a query when category is not defined.
```suggestion
        // decide if the code was changed
        $existing_cat_id = $this->fields['itilcategories_id'] ?? 0;
        if ($existing_cat_id > 0 && $category = ITILCategory::getById($existing_cat_id)) {
```